app/src/main/java/net/gsantner/opoc/util/CoolExperimentalStuff.java

```diff
@@ -141,4 +143,44 @@ public void run() {
             w.setLayout(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.MATCH_PARENT);
         }
     }
+
+    private static Thread autoScrollerThread = null;
+
+    public static void startAutoScroller(final Activity activity, final boolean isWebView) {
+        View view;
+        if (isWebView) {
+            view = activity.findViewById(R.id.document__fragment_view_webview);
+        } else {
+            view = activity.findViewById(R.id.document__fragment__edit__content_editor__scrolling_parent);
+        }
+
+        final int scrollAmount = view.getHeight() / 3;
+        final int millisecondsDelay = 5000;
+
+        autoScrollerThread = new Thread() {
+            @Override
+            public void run() {
+                try {
+                    while (view.getVisibility() == View.VISIBLE) {
+                        Thread.sleep(millisecondsDelay);
+                        view.scrollBy(0, scrollAmount);
+                    }
+                } catch (Exception ignored) {
+                }
+            }
+        };
+        autoScrollerThread.start();
+    }
+
+    public static void stopAutoScrollerIfExists() {
```

Comment:
I think it would be fine to set a static bool flag to false, and at the loop check for that flag.
Then its nearly impossible to run into exception, thus less error prone.